services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/TopicCleanupService.java

```diff
@@ -235,40 +239,48 @@ Admin getAdmin() {
    * If version topic deletion takes more than certain time it refreshes the entire topic list and start deleting from RT topics again.
     */
   void cleanupVeniceTopics() {
+    // TODO - instead of adding all topics into a priority queue we should simply use two separate normal queues
+    // for RT and nonRT topics; this will save a lot of time wasted in sorting
     PriorityQueue<PubSubTopic> allTopics = new PriorityQueue<>(topicPriorityComparator);
     populateDeprecatedTopicQueue(allTopics);
     topicCleanupServiceStats.recordDeletableTopicsCount(allTopics.size());
     long refreshTime = System.currentTimeMillis();
+
     while (!allTopics.isEmpty()) {
       PubSubTopic topic = allTopics.poll();
+      String storeName = topic.getStoreName();
       try {
-        if (topic.isRealTime() && !multiDataCenterStoreToVersionTopicCount.isEmpty()) {
-          // Only delete realtime topic in child fabrics if all version topics are deleted in all child fabrics.
-          if (isRTTopicDeletionBlocked) {
+        String clusterDiscovered;
+        try {
+          clusterDiscovered = admin.discoverCluster(storeName).getFirst();
+        } catch (VeniceNoStoreException e) {
+          LOGGER.warn("Store {} is not found", storeName);
+          getTopicManager().ensureTopicIsDeletedAndBlockWithRetry(topic);
+          topicCleanupServiceStats.recordTopicDeleted();
+          continue;
+        }
+        Store store = admin.getStore(clusterDiscovered, storeName);
+        if (topic.isRealTime() && (this.isRTTopicDeletionBlocked
+            || !admin.isRTTopicDeletionPermittedByAllControllers(clusterDiscovered, store))) {
```

Comment:
In our old implementation, if a topic cannot be deleted (canDelete = false), it will continue and bypass the code from L278 - L289, but in the new version, the bypassing logic is missing. is this on purpose?